policy/pa.go

```diff
@@ -111,6 +112,42 @@ func (pa *AuthorityImpl) loadHostnamePolicy(b []byte) error {
 	return nil
 }
 
+// SetChallengesWhitelistFile will load the given whitelist file, returning error if it
+// fails. It will also start a reloader in case the file changes.
+func (pa *AuthorityImpl) SetChallengesWhitelistFile(f string) error {
+	_, err := reloader.New(f, pa.loadChallengesWhitelist, pa.challengesWhitelistLoadError)
+	return err
+}
+
+func (pa *AuthorityImpl) challengesWhitelistLoadError(err error) {
+	pa.log.AuditErr(fmt.Sprintf("error loading challenges whitelist: %s", err))
+}
+
+func (pa *AuthorityImpl) loadChallengesWhitelist(b []byte) error {
+	hash := sha256.Sum256(b)
+	pa.log.Info(fmt.Sprintf("loading challenges whitelist, sha256: %s",
+		hex.EncodeToString(hash[:])))
+	var wl map[string][]int64
+	err := json.Unmarshal(b, &wl)
+	if err != nil {
+		return err
+	}
+
+	chalWl := make(map[string]map[int64]bool)
+
+	for k, v := range wl {
+		chalWl[k] = make(map[int64]bool)
+		for _, i := range v {
+			chalWl[k][i] = true
+		}
+	}
+
+	// whitelist is read atomically, no locking needed.
```

Comment:
We do actually need locking here, and where we check the list, in `ChallengeTypeEnabled`. Feel free to reuse `blacklistMu` for this purpose. Here it should be `blacklistMu.Lock()` / `.Unlock()`. In `ChallengeTypeEnabled` it should be `blacklistMu.RLock()`.

Reply:
done